hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/JavaDefaultSerializers.java

```diff
@@ -290,4 +303,36 @@ public void destroy() {
     private JavaDefaultSerializers() {
     }
 
+    /**
+     * Utility class to access internal inflater of GZIPInputStream
+     */
+    private static class ExtendedGZipInputStream extends GZIPInputStream {
+
+        static final int GZIP_TRAILER_SIZE = 8;
+
+        ExtendedGZipInputStream(InputStream in) throws IOException {
+            super(in);
+        }
+
+        Inflater getInflater() {
+            return inf;
+        }
+    }
+
+    /**
+     * Gzip input stream reads more bytes in the stream than it actually needs.
+     * This method adjust the position so that, next data in the stream can be read correctly
+     */
+    private static void rewindStream(ObjectDataInput in, ExtendedGZipInputStream gZipInputStream) {
```

Comment:
why is this method static anyway? perhaps you don't need to expose the inflater at all and have the logic nicely encapsulated inside this class. something along this lines `pushBackUnconsumedBytes()`. just an idea anywya.